### PR TITLE
Extended CloudFormation section

### PR DESCRIPTION
**Pull request by gscalise**
Hi!, I've added some tips for CloudFormation based on my experience with it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1090,20 +1090,24 @@ CloudFormation
...
 ### CloudFormation Alternatives and Lock-In
 
 -	Hashicorp’s [Terraform](https://www.terraform.io/intro/vs/cloudformation.html) is a third-party alternative.
+	-	🔸 Some AWS features may not be available in Terraform (ie., MultiAZ ElastiCache using Redis), and you may have to resort to embedded **CloudFormation** templates.
 
 ### CloudFormation Tips
 
 -	[Troposphere](https://github.com/cloudtools/troposphere) is a Python library that makes it much easier to create CloudFormation templates.
+-	If you are building different stacks with similar layers, it may be useful to build separate templates for each layer that you can reuse using [AWS::CloudFormation::Stack](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stack.html).
+	-	🔸 Avoid hardcoding resource parameters that can potentially change. Use stack parameters as much as you can, and resort to default parameter values.
```
Comment (gscalise) [inline README.md]: Fixed now ;)

Comment (ThanosBaskous) [inline README.md]: Consistency nit: The leading marker doesn't have a space after it throughout the rest of the guide.

**Comment by ThanosBaskous** [comment]
Thanks for your submission, @gscalise! I have one small nit.

File: README.md (hunk 1 above)
Comment (gscalise) [inline README.md]: The space is now removed.

File: README.md [hunk 2]
```diff
@@ -1090,20 +1090,24 @@ CloudFormation
...
 
 -	📒 [Homepage](https://aws.amazon.com/cloudformation/) ∙ [Developer guide](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/) ∙ [FAQ](https://aws.amazon.com/cloudformation/faqs/) ∙ [Pricing](https://aws.amazon.com/cloudformation/pricing/) at no additional charge
--	**CloudFormation** offers mechanisms to create and manage entire configurations of many types of AWS resources, using a JSON-based templating language.
+-	**CloudFormation** offers mechanisms to create and manage entire configurations of many types of AWS resources, using a templating language based on JSON (🔹and YAML since Q3 2016). 
 -	💸CloudFormation itself has [no additional charge](https://aws.amazon.com/cloudformation/pricing/) itself; you pay for the underlying resources.
 
 ### CloudFormation Alternatives and Lock-In
 
 -	Hashicorp’s [Terraform](https://www.terraform.io/intro/vs/cloudformation.html) is a third-party alternative.
+	-	🔸Some AWS features may not be available in Terraform (ie., MultiAZ ElastiCache using Redis), and you may have to resort to embedded **CloudFormation** templates.
```
Comment (ThanosBaskous) [inline README.md]: nit:i.e.

File: README.md [hunk 3]
```diff
@@ -1090,20 +1090,24 @@ CloudFormation
...
+-	If you are building different stacks with similar layers, it may be useful to build separate templates for each layer that you can reuse using [AWS::CloudFormation::Stack](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stack.html).
+-	🔸Avoid hardcoding resource parameters that can potentially change. Use stack parameters as much as you can, and resort to default parameter values.
 -	🔹Until [2016](https://aws.amazon.com/about-aws/whats-new/2016/09/aws-cloudformation-introduces-yaml-template-support-and-cross-stack-references/), CloudFormation used only an awkward JSON format that makes both reading and debugging difficult. To use it effectively typically involved building additional tooling, including converting it to YAML, but now [this is supported directly](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-formats.html).
 
 ### CloudFormation Gotchas and Limitations
 
+-	❗Modifications to stack resources made outside CloudFormation can potentially lead to stacks stuck in UPDATE\_ROLLBACK\_FAILED mode. Stacks in this state can't be recovered without help from AWS Support.
```
Comment (ThanosBaskous) [inline README.md]: nit: We use unicode quote characters (’ instead of ') throughout the guide.

Comment (gscalise) [inline README.md]: Hi! The two issues have been amended.

**Comment by ThanosBaskous** [comment]
LGTM. Thanks @gscalise! Bouncing to @jlevy for final review since CloudFormation doesn't yet have a topic editor.

File: README.md [hunk 4]
```diff
@@ -1090,20 +1090,24 @@ CloudFormation
...
 
 -	📒 [Homepage](https://aws.amazon.com/cloudformation/) ∙ [Developer guide](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/) ∙ [FAQ](https://aws.amazon.com/cloudformation/faqs/) ∙ [Pricing](https://aws.amazon.com/cloudformation/pricing/) at no additional charge
--	**CloudFormation** offers mechanisms to create and manage entire configurations of many types of AWS resources, using a JSON-based templating language.
+-	**CloudFormation** offers mechanisms to create and manage entire configurations of many types of AWS resources, using a templating language based on JSON (🔹and YAML since Q3 2016). 
 -	💸CloudFormation itself has [no additional charge](https://aws.amazon.com/cloudformation/pricing/) itself; you pay for the underlying resources.
 
 ### CloudFormation Alternatives and Lock-In
 
 -	Hashicorp’s [Terraform](https://www.terraform.io/intro/vs/cloudformation.html) is a third-party alternative.
+	-	🔸Some AWS features may not be available in Terraform (i.e. MultiAZ ElastiCache using Redis), and you may have to resort to embedded **CloudFormation** templates.
```
Comment (jlevy) [inline README.md]: Punctuation nits:

> (e.g. multi-AZ ElastiCache using Redis)

Comment (jlevy) [inline README.md]: Nit: Remove boldface on **CloudFormation** as it's already been bolded.

**Comment by gscalise** [comment]
Hi! I've just finished merging my changes to the current commit in the master branch.

**Comment by ThanosBaskous** [comment]
Thanks @gscalise!

**Pull request merged**